src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -1104,17 +1105,53 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 		if (!this.xterm?.raw.element) {
 			return;
 		}
+		this._bufferElementProvider?.dispose();
 		this._accessibilityBuffer = this.xterm.raw.element.querySelector('.xterm-accessibility-buffer') as HTMLElement || undefined;
 		if (!this._accessibilityBuffer) {
 			return;
 		}
-		const accessibleBufferContentEditable = this._configurationService.getValue(TerminalSettingId.AccessibleBufferContentEditable);
+		const accessibleBufferContentEditable = isLinux ? 'on' : this._configurationService.getValue(TerminalSettingId.AccessibleBufferContentEditable);
 		this._accessibilityBuffer.contentEditable = accessibleBufferContentEditable === 'on' || (accessibleBufferContentEditable === 'auto' && !this._accessibilityService.isScreenReaderOptimized()) ? 'true' : 'false';
 		// The viewport is undefined when this is focused, so we cannot get the cell height from that. Instead, estimate using the font.
 		const font = this.xterm.getFont();
 		const lineHeight = font?.charHeight ? font.charHeight * font.lineHeight + 'px' : '';
 		this._accessibilityBuffer.style.lineHeight = lineHeight;
+		const commands = this.capabilities.get(TerminalCapability.CommandDetection)?.commands;
+		if (!commands?.length) {
+			const noContent = document.createElement('div');
+			const noContentLabel = nls.localize('terminal.integrated.noContent', "No terminal content available for this session.");
+			noContent.ariaLabel = noContentLabel;
+			noContent.textContent = noContentLabel;
+			const fragment = document.createDocumentFragment();
+			fragment.appendChild(noContent);
+			this._bufferElementProvider = this._register(this.xterm.raw.registerBufferElementProvider({ provideBufferElements: () => fragment }));
+			this._accessibilityBuffer.focus();
+			return;
+		}
+		const shellIntegrationFragment = document.createDocumentFragment();
+		let header;
+		for (const command of commands) {
+			header = document.createElement('h2');
+			// without this, the text area gets focused when keyboard shortcuts are used
+			header.tabIndex = -1;
+			header.textContent = command.command.replace(new RegExp(' ', 'g'), '\xA0');
+			const output = document.createElement('div');
+			// without this, the text area gets focused when keyboard shortcuts are used
+			output.tabIndex = -1;
+			output.textContent = command.getOutput()?.replace(new RegExp(' ', 'g'), '\xA0') || '';
+			shellIntegrationFragment.appendChild(header);
+			shellIntegrationFragment.appendChild(output);
+		}
+		if (header) {
+			// focus the cursor line's header
+			header.tabIndex = 0;
+		}
+		this._bufferElementProvider = this._register(this.xterm.raw.registerBufferElementProvider({ provideBufferElements: () => shellIntegrationFragment }));
```

Comment:
Can we register this once when xterm is opened?

Reply:
we won't have the elements at that time so don't think so

Reply:
neat 😄